src/server/slowlog.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <boost/circular_buffer.hpp>
+#include <string>
+#include <vector>
+
+#include "base/integral_types.h"
+#include "server/common.h"
+namespace dfly {
+
+class SlowlogEntry {
+ public:
+  SlowlogEntry(uint64_t command_id, uint64_t unix_timestamp, uint64_t execution_time_micro,
+               std::vector<std::string> cmd_args, std::string client_ip, std::string client_name);
+
+  const std::vector<std::string>& cmdArgs() const {
+    return cmd_args_;
+  }
+  int64 commandId() const {
+    return command_id_;
+  }
+  int64 unixTimestamp() const {
+    return unix_timestamp_;
+  }
+  int64 executionTimeMicro() const {
+    return execution_time_micro_;
+  }
+  const std::string& clientIp() const {
+    return client_ip_;
+  }
+  const std::string& clientName() const {
+    return client_name_;
+  }
+
+ private:
+  std::vector<std::string> cmd_args_;
+  uint64_t command_id_;
+  uint64_t unix_timestamp_;
+  uint64_t execution_time_micro_;
+  std::string client_ip_;
+  std::string client_name_;
+};
+
+class SlowlogData {
+ public:
+  SlowlogData() = default;
+  SlowlogData(int64 slowlogEntryId, int totalThreads)
+      : slowlog_entry_id_(slowlogEntryId), total_threads_(totalThreads) {
+    slowlog_ = boost::circular_buffer<SlowlogEntry>(0);
+  };
+  boost::circular_buffer<SlowlogEntry>& Slowlog();
+  void Add(std::string& command_name, absl::Span<absl::Span<char>>& args, std::string client_name,
```

Comment:
avoid passing mutable references as input arguments - use `const string&` for example.
Span should be passed by value. client_name should be passed as `const string&`

Reply:
changed